src/controllers/osd.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var xml = require('xml');
+var nconf = require('nconf');
+
+var plugins = require('../plugins');
+var meta = require('../meta');
+
+module.exports = function (req, res, next) {
+	if (plugins.hasListeners('filter:search.query')) {
+		res.type('application/xml').send(generateXML());
+	} else {
+		next();
+	}
+};
```

Comment:
Please have the function export an object with the middleware above as a property of said object. I'd recommend `handle` as the method name, so below, you can `app.get('/osd.xml', controllers.osd.handle);`